server/plugin/command.go

```diff
@@ -503,6 +553,158 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
 	}
 }
 
+func (p *Plugin) handleWebhookAdd(ctx context.Context, parameters []string, githubClient *github.Client) string {
+	if len(parameters) < 1 {
+		return "Invalid parameter for add command, provide repo details in `owner[/repo]` format."
+	}
+
+	siteURL := p.getSiteURL()
+	if strings.Contains(siteURL, "localhost") {
+		return fmt.Sprintf("Your Site URL should be a public URL. You can configure the Site URL [here](%s/admin_console/environment/web_server)", *p.API.GetConfig().ServiceSettings.SiteURL)
+	}
+	baseURL := p.getBaseURL()
+	owner, repo := parseOwnerAndRepo(parameters[0], baseURL)
+	var hook github.Hook
+	var config = make(map[string]interface{})
+	config["secret"] = p.getConfiguration().WebhookSecret
+	config["url"] = siteURL + "/plugins/" + Manifest.Id + "/webhook"
+	config["insecure_ssl"] = false
+	config["content_type"] = "application/json"
+	hook.Events = []string{"*"}
+
+	if len(parameters) == 2 {
+		if strings.Contains(parameters[1], "*") && len(parameters[1]) == 1 {
+			hook.Events = []string{"*"}
+		}
+		if err := p.checkOptionsValid(strings.Split(parameters[1], ",")); err != nil {
+			return err.Error()
+		}
+		hook.Events = strings.Split(parameters[1], ",")
+	}
+	hook.Config = config
+	githubHook, _, err := p.CreateHook(ctx, githubClient, owner, repo, hook)
+	if err != nil {
+		return err.Error()
+	}
+
+	hookDetails, _, err := p.GetHook(ctx, githubClient, owner, repo, *githubHook.ID)
+	if err != nil {
+		return err.Error()
+	}
+
+	hookURL := baseURL + owner
+	if repo != "" {
+		hookURL += "/" + repo
+	}
+	hookURL += githubHookURL
+	txt := "Webhook Created Successfully \n"
+	hookID := strconv.Itoa(int(*githubHook.ID))
+	txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hookDetails.Events, " , "))
+	return txt
+}
+func (p *Plugin) checkOptionsValid(options []string) error {
+	for _, val := range options {
+		isValidEvent := false
+		for _, item := range events {
+			if item == val {
+				isValidEvent = true
+				break
+			}
+		}
+		if !isValidEvent {
+			return errors.New(val + " is not a valid events to trigger webhook")
+		}
+	}
+	return nil
+}
+func (p *Plugin) handleWebhookList(ctx context.Context, parameters []string, githubClient *github.Client) string {
+	if len(parameters) == 0 {
+		return "Invalid parameter for list command, provide repo details in `owner[/repo]` format."
+	}
+	if len(parameters) != 1 {
+		return "Invalid parameter for list command, only accept one argument."
+	}
+	baseURL := p.getBaseURL()
+	owner, repo := parseOwnerAndRepo(parameters[0], baseURL)
+	var githubHookList []*github.Hook
+	var err error
+	opt := &github.ListOptions{
+		PerPage: 50,
+	}
+	txt := "### Webhooks in this Repository\n"
```

Comment:
Nit pick: Move this assignment  into L639.

Reply:
@hanzei done